src/main/java/io/microprofile/config/Config.java

```diff
@@ -13,418 +13,70 @@
-     * @param defaultValue The default value.
-     *
-     * @return The associated BigDecimal if propertyName is found and has valid
-     *         format, default value otherwise.
-     */
-    BigDecimal getBigDecimal(String propertyName, BigDecimal defaultValue);
-
-    /**
-     * Get a {@link BigInteger} associated with the given configuration propertyName or {@code null} if the property does not exist.
-     *
-     * @param propertyName The configuration propertyName.
-     *
-     * @return The associated BigInteger if propertyName is found and has valid format or {@code null} if the property does not exist.
-     */
-    BigInteger getBigInteger(String propertyName);
-
-    /**
-     * Get a {@link BigInteger} associated with the given configuration propertyName.
-     * If the propertyName doesn't map to an existing object, the default value
-     * is returned.
-     *
-     * @param propertyName The configuration propertyName.
-     * @param defaultValue The default value.
-     *
-     * @return The associated BigInteger if propertyName is found and has valid
-     *         format, default value otherwise.
+     *         is not a String.
      */
-    BigInteger getBigInteger(String propertyName, BigInteger defaultValue);
+    Optional<String> getString(String propertyName);
 
-    /**
-     * Get a string associated with the given configuration propertyName or {@code null} if the property does not exist.
-     *
-     * @param propertyName The configuration propertyName.
-     * @return The associated string or {@code null} if the property does not exist.
-     *
-     * @throws IllegalArgumentException is thrown if the propertyName maps to an object that
-     *         is not a String.
+    
+    
+	/**
+	 * Return a collection of property names
+	 * @return the property names
+	 */
+	Collection<String> 	getPropertyNames();
+	/**
+     * Filter the configured value.
+     * This can e.g. be used for decryption.
+     * @return the filtered value
      */
-    String getString(String propertyName);
+    String filterConfigValue(String key, String value);
 
     /**
-     * Get a URL associated with the given configuration propertyName.
-     * If the propertyName doesn't map to an existing object, the default value
-     * is returned.
-     *
-     * @param propertyName The configuration propertyName.
-     * @param defaultValue The default value.
-     * @return The associated URL if propertyName is found and has valid
-     *         format, default value otherwise.
-     *
-     * @throws IllegalArgumentException is thrown if the propertyName maps to an object that
-     *         is not a URL.
+     * Filter the configured value for logging.
+     * This can e.g. be used for displaying ***** instead of a real password.
+     * @return the filtered value
      */
-    URL getURL(String propertyName, URL defaultValue);
-	
+    String filterConfigValueForLog(String key, String value);
```

Comment:
I can at least explain what it originally did in my own proposal ;) 
In DeltaSpike I had the problem to output config values into the log and to some admin UI. E.g. we log out all picked up values at startup. And also if they dynamically change during runtime. 
And of course you don't like to output credentials and other secret stuff to the log and neither on some admin UI. Logs *must not* create credentials and stuff! 
But the only one who knows what to star-out is the developer. So I implemented the possibility to add a pluggable ConfigFilter for filtering values before they get logged (or create any other readable output) based on the key.